streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -34,77 +35,19 @@
     private final Map<String, String> storeToChangelogTopic;
     private final Set<String> repartitionTopics;
 
-    public static ProcessorTopology with(final List<ProcessorNode> processorNodes,
```

Comment:
These factories were only used by tests, so it's not appropriate to have them in the main module. Their presence here made it more difficult to analyse the production code paths that construct a ProcessorTopology.

So, I relocated these factory methods to the test module. The ones that were used in exactly one test got moved to that test class, and the others got moved to a `ProcessorTopologyFactories` test utility.

Reply:
+100